sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricName.java

```diff
@@ -48,12 +53,24 @@ public String toString() {
   public static MetricName named(String namespace, String name) {
     checkArgument(!Strings.isNullOrEmpty(namespace), "Metric namespace must be non-empty");
     checkArgument(!Strings.isNullOrEmpty(name), "Metric name must be non-empty");
-    return new AutoValue_MetricName(namespace, name);
+    return new AutoValue_MetricName(namespace, name, new HashMap<String, String>());
+  }
+
+  public static MetricName named(String namespace, String name, HashMap<String, String> labels) {
```

Comment:
change the map type to immutablemap throughout (change to ImmutableMap.of() for the empty ones)

Nice to have immutable to avoid possible changes to the labels. And it is optimized for small maps